File: frontend/lib/features/habits/data/models/requests/habit_participation.dart
class HabitParticipationUpdateRequestModel {
  final String color;
  final bool toGain;
  final bool notificationsReminderEnabled;
  final String? reminderTime;
  final String? reminderBody;

  const HabitParticipationUpdateRequestModel({
    required this.color,
    required this.toGain,
    required this.notificationsReminderEnabled,
    required this.reminderTime,
    required this.reminderBody,
  });

  Map<String, dynamic> toJson() {
    return {
      'color': color,
      'to_gain': toGain,
      'notifications_reminder_enabled': notificationsReminderEnabled,
      'reminder_time': reminderTime,
      'reminder_body': reminderBody,
    };
  }
}

class HabitParticipationCreateRequestModel {
  final String habitId;
  final String color;
  final bool toGain;

  const HabitParticipationCreateRequestModel({
    required this.habitId,
    required this.color,
    required this.toGain,
  });

  Map<String, dynamic> toJson() {
    return {
      'habit_id': habitId,
      'color': color,
      'to_gain': toGain,
    };
  }
}

File: backend/src/features/habits/structs/requests/habit_participation.rs
use chrono::NaiveTime;
use serde::{Deserialize, Serialize};
use uuid::Uuid;

#[derive(Deserialize)]
pub struct GetHabitParticipationParams {
    pub habit_participation_id: Uuid,
}

#[derive(Deserialize)]
pub struct UpdateHabitParticipationParams {
    pub habit_participation_id: Uuid,
}

#[derive(Deserialize, Serialize)]
pub struct HabitParticipationUpdateRequest {
    pub color: String,
    pub to_gain: bool,
    pub notifications_reminder_enabled: bool,
    pub reminder_time: Option<NaiveTime>,
    pub reminder_body: Option<String>,
}

#[derive(Deserialize)]
pub struct HabitParticipationCreateRequest {
    pub habit_id: uuid::Uuid,
    pub color: String,
    pub to_gain: bool,
}
